docs/resources/model_serving.md

```diff
@@ -11,17 +11,17 @@ This resource allows you to manage [Model Serving](https://docs.databricks.com/m
 resource "databricks_model_serving" "this" {
   name = "ads-serving-endpoint"
   config {
-    served_models {
+    served_entities {
```

Comment:
We cannot do this directly right? There has to be a deprecation pathway otherwise it would be a breaking change.

Reply:
+ @nkvuong

Reply:
you need a state migration - similar to this https://github.com/databricks/terraform-provider-databricks/blob/08f816dd93ae969a0e2820386af2d19bc8c5bfde/catalog/resource_metastore_data_access.go#L81

Reply:
@nkvuong do we really need it if the people can just change from `served_models` to `served_entities`? It should work imho

Reply:
I agree with Alex. Apart from the name changes:
`served_models` -> `served_entities`
`served_models.*.model_name` -> `served_entities.*.entity_name`
`served_models.*.model_version` -> `served_entities.*.entity_version`

And the addition of `served_entities.*.external_model`, the two resources are interchangeable.

We still provide the option to use `served_models`, but a deprecation message will be added stating that you should use `served_entities`.

Reply:
the issue will be with the read method, as it will return both `served_models` and `served_entities` and Terraform will treat that as a diff, so we need a way to suppress that.

Reply:
FWIW, we don't store separate state for `served_models` and `served_entities`, and only one can ever be populated at a time on an endpoint. So what we can do is when both are returned, we discard the one that the user did not specify and populate the one that the user did (is that possible in Terraform)? @nkvuong

Reply:
@arpitjasa-db That's what the latest changes are about - discarding what isn't specified by user.
One question - if I replace `served_models` with `served_entities` in the existing serving endpoint - it won't cause any redeployment, etc.?

Reply:
@mgyucht ^

Reply:
Well it doesn't cause a real redeployment, but it behaves the same as when you call "UpdateEndpoint" with no diff in the config, where it go into an updating state briefly (~30 seconds) and increments the config version
<img width="1532" alt="Screenshot 2024-02-05 at 4 03 10 PM" src="https://github.com/databricks/terraform-provider-databricks/assets/87999496/48ac8963-3933-45a4-9acc-759cfed8e88d">

compared to a real update which takes longer (~15 min)
<img width="1513" alt="Screenshot 2024-02-05 at 4 57 50 PM" src="https://github.com/databricks/terraform-provider-databricks/assets/87999496/8c7e3789-7855-4d3f-ab57-affd8b860c32">

Reply:
Thank you! @840 We'll need to put a note saying about that